From f4207b8314fa88f60b0e337055f15f3525d557cc Mon Sep 17 00:00:00 2001
From: Joergen Ibsen <ji@ibse.dk>
Date: Tue, 31 Oct 2017 09:59:29 +0100
Subject: [PATCH] Fix preprocessor check for inline asm

With MSVC inline asm is only available when compiling for x86.
WIN32 is defined for all of x86, x64, and ARM. This checks for
MSVC and x86 instead.
---
 src/platform.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/platform.cpp b/src/platform.cpp
index 492053ae7..248ae6e26 100644
--- a/src/platform.cpp
+++ b/src/platform.cpp
@@ -64,7 +64,7 @@ static inline unsigned long read_time(void)
 	asm volatile("mftb %0" : "=r" (a));
 	return a;
 }
-#  elif defined(WIN32)
+#  elif defined(_MSC_VER) && defined(_M_IX86)
 static unsigned __int64 read_time(void)
 {
 	unsigned l, h;
